matching/matcher.go

```diff
@@ -131,3 +132,35 @@ func filterPlacesOnPriceLevel(places []Place, level POI.PriceLevel) []Place {
 	}
 	return results
 }
+
+// Filter parameters related with user rating
+type UserRatingFilterParams struct {
+	MinUserRatings int
+}
+
+// Filter to remove low user rating
+type FilterLowUserRating struct {
+}
+
+// Implement Match from the `Matcher` interface
+func (m FilterLowUserRating) Match(req *FilterRequest) ([]Place, error) {
+	var results []Place
+	filterParams := req.Params[req.Criteria]
+
+	if _, ok := filterParams.(UserRatingFilterParams); !ok {
+		return results, errors.New("user rating matcher received wrong filter params")
+	}
+	params := filterParams.(UserRatingFilterParams)
+	return filterUserRating(req.Places, params.MinUserRatings), nil
+}
+
+func filterUserRating(places []Place, minUserRating int) []Place {
```

Comment:
Consider use the generic filter function in iowrappers/utils.go

Reply:
It seems the generic filter function will return a different type of Place (iowrappers.Place). 
This function here requires matching.Place, so we need to use Generics there?

Reply:
We should refactor this function to accept `interface` and move it to `utils` package.

Reply:
Done. I changed it to use Filter() from iowrappers/utils.go